src/operator/matrix_op-inl.h

```diff
@@ -643,6 +654,10 @@ void Slice(const TBlob &src,
   mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
   int ndim = static_cast<int>(ret->shape_.ndim());
 
+  int axis_size = static_cast<int>(src.shape_[param.axis]);
+  param.begin = (param.begin < 0) ? (param.begin + axis_size + 1) : param.begin;
```

Comment:
But how do you do `1...n-1` without specifying the length explicitly, via `being=1, end=n`? Avoiding this is the whole point of this PR, so that when you are unrolling symbols with different sequence-lengths, you don't need to explicitly inject the new sequence length into every slice layer.

This sort of syntax is also supported by the TensorFlow [slice operator](https://www.tensorflow.org/versions/r0.11/api_docs/python/array_ops.html#slice).